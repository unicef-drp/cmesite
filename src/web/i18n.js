import { addLocaleData } from 'react-intl';
import enLocaleData from 'react-intl/locale-data/en';
import enTranslationMessages from './translations/en.json';

export const APP_SUPPORTED_LANGUAGES = ['en'];
export const DEFAULT_LOCALE = 'en';

addLocaleData(enLocaleData);

export const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE
      ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages)
      : {};
  return Object.keys(messages).reduce((formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE
        ? defaultFormattedMessages[key]
        : messages[key];
    return Object.assign(formattedMessages, { [key]: formattedMessage });
  }, {});
};

export const translationMessages = {
  en: formatTranslationMessages('en', enTranslationMessages),
};
