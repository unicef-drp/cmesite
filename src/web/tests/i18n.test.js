import { translationMessages } from '../i18n';

describe('app | i18n', () => {
  it('should ', () => {
    expect(translationMessages).toMatchSnapshot();
  });
});
