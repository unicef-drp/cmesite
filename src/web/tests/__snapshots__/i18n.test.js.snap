// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app | i18n should  1`] = `
Object {
  "en": Object {},
}
`;
